src/commands/deleteCommand.ts

```diff
@@ -0,0 +1,21 @@
+import { Command } from "@yuudachi/framework";
+import type { InteractionParam, ArgsParam, CommandMethod } from "@yuudachi/framework/types";
+import { PermissionFlagsBits } from "discord.js";
+import type { DeleteCommandContextCommand } from "../interactions/context/deleteCommand.js";
+
+export default class extends Command<typeof DeleteCommandContextCommand {
+	public constructor() {
+		super(["Delete Command"]);
+	}
+	public override async messageContext(
+		interaction: InteractionParam<CommandMethod.MessageContext>,
+		args: ArgsParam<typeof DeleteCommandContextCommand>,
+	): Promise<void> {
+	   if(args.message.interaction.user.id !== interaction.user.id || !interaction.member.permissions.has(PermissionFlagsBits.ManageMessages)){
+              interaction.reply({ content: "You are not author of this command.", ephemeral: true })
+              return
+	   }
+           args.message.delete()
+           interaction.reply({ content: "Command successfully deleted.", ephemeral: true })
```

Comment:
Might be wise to defer here? 🤷 
```suggestion
           await args.message.delete()
           await interaction.reply({ content: "Command successfully deleted.", ephemeral: true })
```

Reply:
I don't think it tooks 3 secs to delete

Reply:
It's a good practice to always defer the interaction, the error handler relies on it as well